qiskit/transpiler/passes/synthesis/hls_plugins.py

```diff
@@ -1212,10 +1227,26 @@ def run(self, high_level_object, coupling_map=None, target=None, qubits=None, **
         if not isinstance(high_level_object, ModularAdderGate):
             return None
 
-        if options.get("num_clean_ancillas", 0) >= 1:
-            return adder_ripple_c04(high_level_object.num_state_qubits, kind="fixed")
+        # For up to 5 qubits, the QFT-based adder is best
```

Comment:
How did you determine the "best" here? 🙂

Reply:
A small python script that simply runs all available plugins for each of the adder gates, while varying the number of control and ancillary qubits, and manually choosing the method that minimizes the CX-count. Note that we have an option to automatically run all of the available plugins and choose the decomposition that provides the best possible results. 

Actually, it might be nice to think if the above behavior should be used by default.